pages/create-issue.tsx

```diff
@@ -1,48 +1,101 @@
 import { GetStaticProps } from 'next'
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import BeproService from '../services/bepro';
 import GithubMicroService from '../services/github-microservice';
+import { setLoadingAttributes } from '../providers/loading-provider';
+import { BeproBalance } from '../helpers/bepro-balance';
+import InputNumber from '../components/inputNumber';
 
-export default function PageCreateIssue() {
+interface Amount {
+  value?: string,
+  formattedValue: string,
+  floatValue?: number
+}
 
+export default function PageCreateIssue() {
 
   const [issueTitle, setIssueTitle] = useState("");
   const [issueDescription, setIssueDescription] = useState("");
-  const [issueAmount, setIssueAmount] = useState<number>(0);
+  const [issueAmount, setIssueAmount] = useState<Amount>({ value: '0', formattedValue: '0', floatValue: 0});
+  const [balance, setBalance] = useState<string>('0');
   const [allowedTransaction, setAllowedTransaction] = useState<boolean>(false);
 
+  const useBeproBalance = async () => {
+    setBalance(await BeproBalance())
+  }
+
+  useEffect(() => {
+    useBeproBalance()
+  },[])
```

Comment:
You can set the `balance` state updater directly as a side-effect. Like:

```jsx
 useEffect(() => {
    setBalance(BeproBalance())
  }, [BeproBalance])
```

In the current usage way, `useBeproBalance()` is getting a new reference on every render, getting rerendered too. To avoid this behavior, must memoize it (what is expensive), like:
```jsx
 const useBeproBalance = useCallback(async () => {
    setBalance(await BeproBalance())
 }, [])

useEffect(() => {
    useBeproBalance()
 }, [useBeproBalance])
```

Reply:
Done. -> 98d6e6e. Does this solve?

Reply:
Solves! But was created an unnecessary expensive memoization, as `useBeproBalance()` it's been called just once and by callback hook. By now, in this case, we need to focus on get this BEPRO Staked balance and set it both on **create-issue mount** and when its own **value change**.  Let's just use the getter module inside the `useEffect()` (so we don't need to define this function as an dependency).

Instead of this:

```jsx
  const useBeproBalance = useCallback(async () => {
    await BeproService.login(); 
    setBalance(await BeproService.network.getBEPROStaked())
  }, [])

  useEffect(() => {
    useBeproBalance()
  },[useBeproBalance])
```

Let's do this:

```jsx
useEffect(() => {
  /* > As we're calling an async function, using this to conditionally update
  *  > component state prevents memory leak.
  */
  let isMounted = true;

  void async function getBeproBalance() {
    await BeproService.login(); 

    if (isMounted) {
      setBalance(BeproService.network.getBEPROStaked());
    }
  }();

  return () => {
    isMounted = false;
  }
}, [])
```
Let me know if makes sense for you.

Reply:
Good. The `isMounted` makes it wait for the component to actually mount. behaves like conventional react componentDidMount, right?

Reply:
done -> e16601a

Reply:
More or less. Complementing, we allow updates trigger only on mounted components. `isMounted` don't stop the mouting/unmounting of a component. This is just a common variable that helps us to handle out-of-order fetch calls. In class components, would be something like:
```jsx
constructor() {
  this.isMounted = false;
}

componentDidMount() {
  this.isMounted = true;

  if (this.isMounted) {
    // do something
  }
}

componentWillUnmount() {
  this.isMounted = false;
}